request.go

```diff
@@ -17,6 +20,7 @@ type Request struct {
 
 func NewRequest() *Request {
 	return &Request{
+		ID:            uuid.New(),
```

Comment:
We should make sure that the `Request` object is always generated using `NewRequest()`, so the ID is always set